src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -939,17 +951,25 @@ private bool ParseFile(string[] args, ref int i, bool noexitSeen)
                     }
                     else if (!string.IsNullOrEmpty(arg) && SpecialCharacters.IsDash(arg[0]))
                     {
-                        Match m = argPattern.Match(arg);
-                        if (m.Success)
+                        int offset = arg.IndexOf(':');
+                        if (offset >= 0)
                         {
-                            int offset = arg.IndexOf(':');
                             if (offset == arg.Length - 1)
                             {
                                 pendingParameter = arg.TrimEnd(':');
                             }
                             else
                             {
-                                _collectedArgs.Add(new CommandParameter(arg.Substring(0, offset), arg.Substring(offset + 1)));
+                                string argValue = arg.Substring(offset + 1);
+                                bool? boolValue = GetBoolValue(argValue);
+                                if (boolValue != null)
```

Comment:
Sorry again -
Can we use `LanguagePrimitives.TryConvertTo` or `LanguagePrimitives.TryConvertArg`?
Can we use TryGetBoolValue pattern?

```c#
                               string argValue1 = arg.Substring(offset + 1); 
                               string argValue0 = arg.Substring(0, offset);
                               if (TryGetBoolValue(argValue1, out bool boolValue))  
                               {  
                                       _collectedArgs.Add(new CommandParameter(argValue0, boolValue));  
                              }  
                               else  
                               {  
                                       _collectedArgs.Add(new CommandParameter(argValue0, argValue1));  
                               }  
```